### ISSUE TITLE
If no featured articles exist, do not show "Related Articles" header at bottom of page

### ISSUE THREAD

**Issue by jwflory** [unicef/inventory-hugo-theme#30]
# Summary

When there are no related articles for a page, do not show the "_Related Articles_" header at the bottom of the page

# Background

**Is the improvement related to a problem? Describe the problem**:

If there are no related articles, the header still appears and can be confusing as to why it is there:

![Screenshot of the Related Articles header appearing on a page when there are no related articles](https://user-images.githubusercontent.com/4721034/139919463-a11edd42-37e3-4313-a19f-d6c21f134684.png "Screenshot of the Related Articles header appearing on a page when there are no related articles")

**What does the improvement look like to you?**:

The header should not appear when there are no featured images.

# Details

This can be done with [Hugo conditionals](https://gohugo.io/templates/introduction/#conditionals) to wrap this part of the template. If there are related articles, then show them; but if there are not any, remove the HTML block from the page.

# Outcome

- Better user experience for someone browsing through different pages
- Better default logic for pages where we do not want to show related articles deliberately

**Comment by mmldco** [comment]
HI @jwflory,

I believe I may have a solution for this. I'm fairly new to Hugo so the solution will definitely need double checking before merging.

**Comment by Yavnikaa** [comment]
Can I take this up?

**Comment by Yavnikaa** [comment]
One thing I want to confirm here is that to test out any changes I make, do I directly start a hugo server here with this repo or do I make it as a submodule in an existing hugo site to test out my changes?

**Comment by jwflory** [comment]
> **_@Yavnikaa wrote…_**
> Can I take this up?

Definitely! This is a good one to get started with, and it will introduce you to the templating syntax used in Hugo layouts.

> **_@Yavnikaa wrote…_**
> One thing I want to confirm here is that to test out any changes I make, do I directly start a hugo server here with this repo or do I make it as a submodule in an existing hugo site to test out my changes?

Good question. I suggest a two-part approach to developing and testing this change:

1. Clone this repo and work on your changes using the `exampleSite` included in the repo. Test your changes with the `exampleSite` theme until you get them right.
2. Once verified with the `exampleSite` in the theme, try cloning the `unicef/inventory` repo with the git submodule. Change the git submodule for the theme to your forked repo of the theme with your changes. You can use `git checkout` to change branches where your changes are. Then, you can run the Hugo server on the UNICEF Open Source Inventory site and verify your changes appear as expected there.

Does this make sense?

If you need some guidance on the second part, you can ask for help here or say hi in our [Matrix room](https://matrix.to/#/#unicef-innovation:matrix.org). If you never used Matrix before, you can see the [getting started page](https://element.io/get-started) for the Element chat client.

**Comment by jwflory** [comment]
FYI: @mmldco submitted a PR that addresses this issue, see #68.

**Comment by mmldco** [comment]
HI @jwflory, I'd like to keep working on this.
